File: ios/runanywhere-sdks-latest/sdk/runanywhere-flutter/packages/runanywhere/lib/public/extensions/format_framework.dart
// SPDX-License-Identifier: Apache-2.0
//
// format_framework.dart — human-readable display name for InferenceFramework.
//
// Pure-Dart proxy for the canonical `rac_inference_framework_display_name` C
// ABI mapping in runanywhere-commons (model_types.cpp). The C string table is
// not exposed through the proto bridge — it would cost a Dart↔native FFI call
// for a static label. Mirror the exact same table in Dart so consumers (UI
// banners, status labels) can resolve a human-readable display name
// synchronously.
//
// The mapping is the source-of-truth pair with:
//   • RN:    `@runanywhere/core/Public/Helpers/formatFramework.ts`
//   • Web:   `@runanywhere/core/Public/Helpers/formatFramework.ts`
//   • Swift: `RAInferenceFramework.displayName` (ModelTypes.swift)
//
// Keep in lock-step with commons when a new framework is added.

import 'package:runanywhere/generated/model_types.pbenum.dart';

const Map<InferenceFramework, String> _kFrameworkDisplayNames = {
  InferenceFramework.INFERENCE_FRAMEWORK_ONNX: 'ONNX Runtime',
  InferenceFramework.INFERENCE_FRAMEWORK_SHERPA: 'Sherpa-ONNX',
  InferenceFramework.INFERENCE_FRAMEWORK_LLAMA_CPP: 'llama.cpp',
  InferenceFramework.INFERENCE_FRAMEWORK_COREML: 'Core ML',
  InferenceFramework.INFERENCE_FRAMEWORK_MLX: 'MLX',
  InferenceFramework.INFERENCE_FRAMEWORK_FOUNDATION_MODELS: 'Foundation Models',
  InferenceFramework.INFERENCE_FRAMEWORK_SYSTEM_TTS: 'System TTS',
  InferenceFramework.INFERENCE_FRAMEWORK_FLUID_AUDIO: 'FluidAudio',
  InferenceFramework.INFERENCE_FRAMEWORK_TFLITE: 'TensorFlow Lite',
  InferenceFramework.INFERENCE_FRAMEWORK_EXECUTORCH: 'ExecuTorch',
  InferenceFramework.INFERENCE_FRAMEWORK_MEDIAPIPE: 'MediaPipe',
  InferenceFramework.INFERENCE_FRAMEWORK_MLC: 'MLC',
  InferenceFramework.INFERENCE_FRAMEWORK_PICO_LLM: 'picoLLM',
  InferenceFramework.INFERENCE_FRAMEWORK_PIPER_TTS: 'Piper TTS',
  InferenceFramework.INFERENCE_FRAMEWORK_SWIFT_TRANSFORMERS:
      'Swift Transformers',
  InferenceFramework.INFERENCE_FRAMEWORK_BUILT_IN: 'Built-in',
  InferenceFramework.INFERENCE_FRAMEWORK_QHEXRT: 'QHexRT',
  InferenceFramework.INFERENCE_FRAMEWORK_NONE: 'None',
};

/// Return the canonical human-readable display name for an [InferenceFramework].
///
/// Mirrors `rac_inference_framework_display_name` from commons
/// (model_types.cpp) so cross-platform UIs render the same label without each
/// app maintaining its own switch table.
///
/// Unknown / unspecified values resolve to `"Unknown"` to match the C default
/// branch, mirroring RN/Web `formatFramework` and Swift
/// `RAInferenceFramework.displayName`.
String formatFramework(InferenceFramework? framework) {
  if (framework == null ||
      framework == InferenceFramework.INFERENCE_FRAMEWORK_UNSPECIFIED ||
      framework == InferenceFramework.INFERENCE_FRAMEWORK_UNKNOWN) {
    return 'Unknown';
  }
  return _kFrameworkDisplayNames[framework] ?? 'Unknown';
}

File: ios/runanywhere-sdks-latest/sdk/runanywhere-react-native/packages/core/src/Public/Helpers/formatFramework.ts
/**
 * formatFramework.ts
 *
 * Delegates to the canonical `rac_inference_framework_display_name` C ABI
 * mapping in runanywhere-commons — the same table Swift's
 * `RAInferenceFramework.displayName` (ModelTypes.swift:187) delegates to.
 * The Nitro method is synchronous (pure table lookup), so consumers (UI
 * banners, status labels) keep resolving labels synchronously; results are
 * memoized to avoid repeated JSI hops from render paths.
 */
import { InferenceFramework } from '@runanywhere/proto-ts/model_types';
import {
  isNativeModuleAvailable,
  requireNativeModule,
} from '../../native/NativeRunAnywhereCore';

const displayNameCache = new Map<InferenceFramework, string>();

/**
 * Return the canonical human-readable display name for an
 * `InferenceFramework`, resolved from the commons C ABI so cross-platform
 * UIs render the same label.
 *
 * Unknown / unspecified values resolve to `"Unknown"` to match the C
 * default branch.
 */
export function formatFramework(
  framework?: InferenceFramework | null
): string {
  if (
    framework == null ||
    framework === InferenceFramework.INFERENCE_FRAMEWORK_UNSPECIFIED ||
    framework === InferenceFramework.UNRECOGNIZED
  ) {
    return 'Unknown';
  }
  const cached = displayNameCache.get(framework);
  if (cached !== undefined) {
    return cached;
  }
  if (!isNativeModuleAvailable()) {
    return 'Unknown';
  }
  const name = requireNativeModule().frameworkDisplayName(framework);
  displayNameCache.set(framework, name);
  return name;
}
